agent/api/eni/eni_test.go

```diff
@@ -129,6 +130,19 @@ func TestGetSubnetGatewayIPv4Address(t *testing.T) {
 	assert.Equal(t, ipv4Gw, testENI.GetSubnetGatewayIPv4Address())
 }
 
+// TestGetLinkName tests the retrieval of ENIs name on the instance.
+func TestGetLinkName(t *testing.T) {
+	ifaces, err := net.Interfaces()
```

Comment:
We should probably be mocking the call to this method

Reply:
@rawahars we may want to create an interface to call `net.Interfaces` so its easier to mock in unit tests. This is an example of using an interface for Docker APIs https://github.com/aws/amazon-ecs-agent/blob/847d9e193325f9b5cb0cad834f02dbd3842129c1/agent/dockerclient/dockerapi/docker_client.go#L102 so we could mock these methods instead of the original ones.

Reply:
We had created such a wrapper earlier. We can reuse that here.

Reply:
For this particular case I think it's simpler to just declare an non-exported variable (alias) that we can change in the tests, 
as opposed to creating an interface and having to inject it and all that.

```golang
// Declaration at package level
var netInterfaces = net.Interfaces

// Invocation 
netInterfaces()

// mocking for tests
netInterfaces = func() ([]Interface, error) { ... }
```